styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -210,6 +216,21 @@ public void close() throws IOException {
     }
   }
 
+  @Override
+  public void restore() {
+    final Retryer<Object> retryer = RetryerBuilder.newBuilder()
+        .retryIfException()
+        .withWaitStrategy(WaitStrategies.exponentialWait())
+        .withStopStrategy(StopStrategies.stopAfterAttempt(3))
+        .build();
+
+    try {
+      retryer.call(Executors.callable(this::tryPollPods));
+    } catch (ExecutionException | RetryException e) {
+      throw new RuntimeException(e);
```

Comment:
The connection might well be setup/configured correctly but connections and requests can and do fail at any time. I would rather crash and retry instead of a best effort `restore` that might lose those completed executions. Also, we do not do best effort reads from bigtable and datastore when restoring state. It seems odd to treat k8s differently.